File: app/src/main/java/de/acando/vinyl/VinylApp.java
package de.acando.vinyl;

import android.app.Application;

import de.acando.vinyl.structure.AppComponent;
import de.acando.vinyl.structure.AppInjector;
import de.acando.vinyl.structure.DaggerAppComponent;

public class VinylApp extends Application {

    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        AppInjector.Companion.init(this);
    }

    private void initAppComponent(MainActivity activity) {
        appComponent = DaggerAppComponent.builder().application(this).activity(activity).build();
    }

    public AppComponent getAppComponent(MainActivity activity) {
        if(appComponent == null) {
            initAppComponent(activity);
        }
        return appComponent;
    }
}

File: app/src/main/java/com/acando/vinyl/VinylApp.kt
package com.acando.vinyl

import android.app.Application
import com.acando.vinyl.structure.AppComponent
import com.acando.vinyl.structure.AppInjector

class VinylApp : Application() {

    private var appComponent : AppComponent = null!!

    override fun onCreate() {
        super.onCreate()
        //appComponent = DaggerAppComponent.builder().application(this).build()
        AppInjector.init(this)
    }

    fun getAppComponent() : AppComponent {
        return appComponent
    }
}